### PR TITLE
Changes to number types.

### PR DESCRIPTION
**Pull request by geosp**
* Introducing new example type number range histogram.
* Number now supports find the best functionality.
* Geo type now guards against 0 results in geocoder.

### CODE REVIEW

**Comment by decrapifier** [comment]
<table>
  <thead>
    <tr>
      <th width="50"></th>
      <th width="100%" data-danger-table="true">Fails</th>
    </tr>
  </thead>
  <tbody><tr>
      <td>:no_entry_sign:</td>
      <td>

  Please add a description to your PR
  </td>
    </tr>
  
<tr>
      <td>:no_entry_sign:</td>
      <td>

  This PR has failing tests. Please alleviate the errors and commit them

<details>
  <summary>should load</summary>
  <code>expected [ Array(32) ] to have the same members as [ Array(31) ]</code>
</details>
  </td>
    </tr>
  </tbody>
</table>


<table>
  <thead>
    <tr>
      <th width="50"></th>
      <th width="100%" data-danger-table="true">Warnings</th>
    </tr>
  </thead>
  <tbody><tr>
      <td>:warning:</td>
      <td>

  The README has not been updated. Please update the README.
  </td>
    </tr>
  </tbody>
</table>


<table>
  <thead>
    <tr>
      <th width="50"></th>
      <th width="100%" data-danger-table="true">Messages</th>
    </tr>
  </thead>
  <tbody><tr>
      <td>:book:</td>
      <td>

  We were able to automatically fix some formatting issues in this PR for you!
  </td>
    </tr>
  </tbody>
</table>


<p align="right">
  Generated by :no_entry_sign: <a href="http://github.com/danger/danger-js/">dangerJS</a>
</p>

File: src/example-types/geo.js [hunk 1]
```diff
@@ -12,13 +12,21 @@ let geo = ({
...
-        let location = response.results[0].geometry.location
+        if (response.error_message) {
+          throw response.error_message
+        }
+
+        let geolocation = _.flow(
+          _.head,
+          _.get('geometry.location'),
+          result => result || { lat: 0, lng: 0 }
+        )(response.results)
```
Comment (sadasant) [inline src/example-types/geo.js]: Nice!

File: src/example-types/number.js [hunk 2]
```diff
@@ -1,108 +1,76 @@
...
-                histogram: {
-                  field,
-                  interval,
-                  min_doc_count: 0,
-                },
-              },
-            },
-          },
+      results = _.extend(results, {
+        extremes: {
```
Comment (daedalus28) [inline src/example-types/number.js]: lets call this `bestRange`

File: README.md [hunk 3]
```diff
@@ -173,18 +173,39 @@ Exists represents whether or not a field is present on results
...
 {
   field: String,
   min: Number,
-  max: Number
+  max: Number,
+  percentileInterval: Number,
+  rangeThreshold: Number,
+  findBestRange = Boolean
+}
+```
```
Comment (daedalus28) [inline README.md]: We should also add documentation on the returned context props

File: CHANGELOG.md [hunk 4]
```diff
@@ -1,4 +1,8 @@
-﻿### 0.6.9
+### 0.6.91
```
Comment (daedalus28) [inline CHANGELOG.md]: This is a new feature so it should be 0.7.0

**Pull request merged**